src/components/profile/ProfileImage.tsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import { AsyncStorage, View, Image, Text, StyleSheet } from "react-native";
+import metrics from "../../config/metrics";
+import { ProfileName } from "./ProfileName";
+import { StateContext } from "../../state";
+
+const styles = StyleSheet.create({
+  container: { alignItems: "center" },
+  image: {
+    width: metrics.DIAMETER,
+    height: metrics.DIAMETER,
+    borderRadius: metrics.RADIUS,
+    borderWidth: 4,
+    borderColor: "#fff",
+    alignSelf: "center"
+  },
+  imageContainer: {
+    width: metrics.DIAMETER,
+    height: metrics.DIAMETER,
+    borderRadius: metrics.RADIUS,
+    alignContent: "center",
+    marginTop: metrics.RADIUS,
+    position: "absolute",
+    shadowColor: "#000000",
+    shadowOpacity: 0.8,
+    shadowRadius: 3,
+    shadowOffset: {
+      height: 5,
+      width: 5
+    },
+    elevation: 10
+  },
+  background: {
+    backgroundColor: "#f5f5f5",
+    height: metrics.DIAMETER,
+    width: "100%",
+    alignItems: "center"
+  },
+  margin: {
+    marginHorizontal: 16,
+    height: 55,
+    borderBottomWidth: 1,
+    borderColor: "#A9A9A9"
+  }
+});
+
+interface ProfileImageProps {
+  recipient: any;
+  navigation: any;
+  isPreview: boolean;
+}
+
+class ProfileImage extends React.Component<ProfileImageProps> {
+  static contextType = StateContext;
+
+  async componentDidMount() {
+    const [, dispatch] = this.context;
+
+    const storedTimeAccepted = await AsyncStorage.getItem(
+      "@storedTimeAccepted"
```

Comment:
```suggestion
      "@storedTimeAccepted"
```
Are you sure you want to persist this? If yes, considering NOT getting it from fs here. Instead rehydrate the context at some point in time into context and then read from context instead of directly from fs. Similar operation as how you load the stored workpass

Reply:
Also, keep this as a separate function, not in componentDidMount.